build/Build.cs

```diff
@@ -136,27 +121,27 @@ static bool HasAuthenticodeSignature(AbsolutePath fileInfo)
         .DependsOn(Restore)
         .Executes(() =>
         {
-            ModifyTemplatedVersionAndProductFilesWithValues(out var versionInfoRestoreAction, out var productWxsRestoreAction);
+            var (versionInfoFile, productWxsFile) = ModifyTemplatedVersionAndProductFilesWithValues();
 
             RuntimeIds.Where(x => x.StartsWith("linux-"))
                 .ForEach(runtimeId => RunBuildFor(NetCore, runtimeId));
             
-            versionInfoRestoreAction.Invoke();
-            productWxsRestoreAction.Invoke();
+            versionInfoFile.Dispose();
+            productWxsFile.Dispose();
         });
 
     [PublicAPI]
     Target BuildOsx => _ => _
         .DependsOn(Restore)
         .Executes(() =>
         {
-            ModifyTemplatedVersionAndProductFilesWithValues(out var versionInfoRestoreAction, out var productWxsRestoreAction);
+            var (versionInfoFile, productWxsFile) = ModifyTemplatedVersionAndProductFilesWithValues();
```

Comment:
I wonder if this would work better if `ModifyTemplatedVersionAndProductFilesWithValues` was 2 separate functions? Then we could use a `using` here and wouldn't need the odd dispose